ospfd/ospfd.c

```diff
@@ -628,15 +627,6 @@ static void ospf_finish_final(struct ospf *ospf)
 	if (ospf->vrf_id == VRF_DEFAULT)
 		ospf_ldp_sync_gbl_exit(ospf, true);
 
-	/* Remove any ospf interface config params */
-	FOR_ALL_INTERFACES (vrf, ifp) {
-		struct ospf_if_params *params;
-
-		params = IF_DEF_PARAMS(ifp);
-		if (OSPF_IF_PARAM_CONFIGURED(params, if_area))
-			UNSET_IF_PARAM(params, if_area);
-	}
-
```

Comment:
I am not sure that anyone was relying on that, because it was not correctly implemented.
Only `ip ospf area X` line (without interface address) was deleted, but all `ip ospf area X A.B.C.D` lines (with interface address) were left in the config.
And there is no mention in the docs that `no router ospf` cleans interface config.

Reply:
we are VMware are having automation where we remove do "no router ospf"  and check for all ospf config going away. I am ok with changes but this was default behaviour and this is changing now may be good idea to document it?

Reply:
It looks more like a changelog to me. Any ideas where to put it in the actual docs?

Reply:
@Spantik I really want these fixes to be merged. Please, point to the place in the docs where you want me to put the notice.

Reply:
@idryzhov you can update docs in doc section. I will get this in for now as document is not blocking.